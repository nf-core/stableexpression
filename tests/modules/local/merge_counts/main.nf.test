nextflow_process {

    name "Test Process MERGE_COUNTS"
    script "modules/local/merge_counts/main.nf"
    process "MERGE_COUNTS"
    tag "merge"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/test_data/merge_counts/input/counts*.csv', checkIfExists: true).collect()
                input[0] = ch_counts
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
