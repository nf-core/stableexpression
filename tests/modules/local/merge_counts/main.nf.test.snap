{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "all_counts.parquet:md5,73784fbc1da2f1c675e04463ab5cc0d7"
                ],
                "1": [
                    [
                        "MERGE_COUNTS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "MERGE_COUNTS",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "counts": [
                    "all_counts.parquet:md5,73784fbc1da2f1c675e04463ab5cc0d7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-21T11:36:51.992454125"
    }
}